Bercea_Bogdan/Week_2/Demo/Program.cs

```diff
@@ -32,22 +37,68 @@ static async Task Main(string[] args)
 
             var result = await interpreter.Interpret(expr, Unit.Default);
 
-            var finalResult = result.Match<bool>(OnRestaurantCreated, OnRestaurantNotCreated);
+            var finalResult = result.Match(OnRestaurantCreated, OnRestaurantNotCreated);
             Assert.True(finalResult);
 
+
+            // User expression
             var userExpr =
                 from userResult in RestaurantDomain.CreateUser("Hanz", "HanzPassword", 45)
                 let user = (userResult as UserCreated)?.User
                 select userResult;
 
             var userRez = await interpreter.Interpret(userExpr, Unit.Default);
 
-            var userFinalResult = userRez.Match<bool>(OnUserCreated, OnUserNotCreated);
+            var userFinalResult = userRez.Match(OnUserCreated, OnUserNotCreated);
             Assert.True(userFinalResult);
 
+            //MenuItem expression
+            var menuItemExpresion =
+                from menuItemResult in RestaurantDomain.CreateMenuItem(
+                    "Pizza", 13.5m, MenuItemValues.getIngredients(), MenuItemValues.getAlergens(), MenuItemValues.getMenu())
+                let menuItem = (menuItemResult as MenuItemCreated)?.MenuItem
+                select menuItemResult;
+
+            var menuItemRez = await interpreter.Interpret(menuItemExpresion, Unit.Default);
+
+            var manuItemFinalResult = menuItemRez.Match(OnMenuItemCreated, OnMenuItemNotCreated);
+            Assert.True(manuItemFinalResult);
+
+            //Employee expression
+            var employeeExpr =
+                from employeeResult in RestaurantDomain.CreateEmployee("Hanz", "Bauer", 30, "Timisoara nr.45",
+                500m, Domain.Models.Gender.Male, Domain.Models.Position.Manager, new Restaurant("Papanasii"))
+                let employee = (employeeResult as EmployeeCreated)?.Employee
+                select employeeResult;
+
+            var employeeRez = await interpreter.Interpret(employeeExpr, Unit.Default);
+
+            var employeeFinalResult = employeeRez.Match(OnEmployeeCreated, OnEmployeeNotCreated);
+            Assert.True(employeeFinalResult);
+
             Console.WriteLine("Hello World!");
```

Comment:
Ok, can you please try to link all of your expressions in order to build a cohesive story? 

You currently have a way to create a restaurant, a menu, a menu item, an employee and an user (which I assume it's a client at this point). If you try to wrap them together you might find some holes that need to be handled.